### DISCUSSION TITLE
Is there a method for partial reset?

### DISCUSSION THREAD

**Discussion by jin60641** [react-hook-form/react-hook-form#2704] [Q&A] [answered]
`formContext.reset()` with no parameters will reset all inputs.
Is it possible to filter the input to be reset by name?

**Comment by bluebill1049** [comment, marked as answer]
```
reset({
   ...getValues(),
  xxxx,
})
```

**Reply by jin60641** [reply]
there's no plan for support new usage like this?
`reset(['name1', 'name2'])`?
Because `formContext.clearErrors()` already support name's array

**Reply by jin60641** [reply]
@bluebill1049

**Reply by bluebill1049** [reply]
no, we don't. it's better to leave those merge work at the user level and then at the lib. if you want to update individual filed then use `setValue`.

**Reply by jin60641** [reply]
is `reset()` method change the defaultValues? if it's right, ...getValues() will change defaultValues.
I can still use  `{...defaultValues, xxxx }` , just a question.

**Reply by bluebill1049** [reply]
`getValues` doesn't return `defaultValues`

**Reply by jin60641** [reply]
Okay Thanks :)

**Reply by bluebill1049** [reply]
no problem, getValues() will only return what's in the form.

**Reply by jin60641** [reply]
I want to reset values and  clear errors by names(partial form).
reset method will clear entire error, if I use setValue, I have to use both setValue and clearErrors, and you said It's best.
Did I get it right?

**Reply by jin60641** [reply]
@bluebill1049 plz check one more time. Thanks. Sorry for annoying

**Reply by bluebill1049** [reply]
no worries @jin60641, I thnk it's better to write a function combine with setValue and clearError and compose those two together.
```
function resetField() {
  setValue();
  clearErrors();
}
```

**Reply by jin60641** [reply]
okay, I'll make it on user level.

**Reply by vijay-rathod-relata** [reply]
Hi, Any updated way to do this, maybe by passing a property in options object of reset function?
